### PR TITLE
Provide relevant error messages

### LINKED ISSUE #23: Feature: provide relevant error messages
Netlify build wraps plugins event handlers with a `try/catch` and report [those as bugs](https://docs.netlify.com/configure-builds/build-plugins/create-plugins/#error-reporting).
Using `catch-all` will result in reporting all errors as user errors (including bugs).

For example we might want to report a relevant error here:
https://github.com/nimbella/netlify-plugin-nimbella/blob/2969462b33aa84d0c44a0c82285dc55e7a439f44/index.js#L84
if the token had been revoked or has expired.

### PR DESCRIPTION
**Pull request by satyarohith**
Tested manually. Everything is working as expected.

Tested review changes.
Closes #23

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -54,39 +56,46 @@ async function deployActions({run, functionsDir, timeout, memory}) {
...
   onPreBuild: async ({utils, constants, inputs}) => {
-    try {
-      if (!process.env.NIMBELLA_LOGIN_TOKEN) {
-        utils.build.failBuild(
-          'Nimbella login token not available. Please run `netlify addons:create nimbella` at the base of your local project directory linked to your Netlify site.'
-        );
-      }
+    if (!process.env.NIMBELLA_LOGIN_TOKEN) {
+      utils.build.failBuild(
+        'Nimbella login token not available. Please run `netlify addons:create nimbella` at the base of your local project directory linked to your Netlify site.'
```
Comment (niks3089) [inline index.js]:
```suggestion
        'Nimbella login token is not available. Please run `netlify addons:create nimbella` at the base of your local project directory linked to your Netlify site.'
```

File: index.js [hunk 2]
```diff
@@ -54,39 +56,46 @@ async function deployActions({run, functionsDir, timeout, memory}) {
...
+    await utils.cache.restore(nimConfig);
 
-      const loggedIn = existsSync(nimConfig);
-      // Login if not logged in before.
-      if (loggedIn) {
-        console.log('\nUsing the following namespace.');
+    const loggedIn = existsSync(nimConfig);
+    // Login if not logged in before.
+    if (loggedIn) {
+      console.log('Using the following namespace.');
```
Comment (niks3089) [inline index.js]: Can we print the console message after `nim auth current` succeeds?

Comment (satyarohith) [reply index.js]: Updated.

File: index.js [hunk 3]
```diff
@@ -54,39 +56,48 @@ async function deployActions({run, functionsDir, timeout, memory}) {
 module.exports = {
   // Execute before build starts.
   onPreBuild: async ({utils, constants, inputs}) => {
-    try {
-      if (!process.env.NIMBELLA_LOGIN_TOKEN) {
+    if (!process.env.NIMBELLA_LOGIN_TOKEN) {
```
Comment (rabbah) [inline index.js]: The logic below checks for cached configs. If they exist, the nimbella token need not exist in the environment - isn't that right? Should the predicate here be `if no token && no cached config`?

Comment (satyarohith) [reply index.js]: > If they exist, the nimbella token need not exist in the environment - isn't that right?

You're right.

**Review by niks3089 — APPROVED** [review]

**Pull request merged**